File: packages/ts/src/code-toolbox/WorkspaceTsCodeToolbox.ts
import { WorkspaceCodeToolbox } from '../Workspace'

export class WorkspaceTsCodeToolbox implements WorkspaceCodeToolbox {
  public getDefaultImage(): string {
    return 'daytonaio/ai-test:0.1.0'
  }

  public getRunCommand(code: string): string {
    const base64Code = Buffer.from(code).toString('base64')
    return `python3 -c "exec(__import__('base64').b64decode('${base64Code}').decode())"`
  }
}

File: packages/python/src/daytona_sdk/code_toolbox/workspace_ts_code_toolbox.py
import base64
from typing import Protocol


class WorkspaceCodeToolbox(Protocol):
    def get_default_image(self) -> str: ...
    def get_run_command(self, code: str) -> str: ...


class WorkspaceTsCodeToolbox:
    def get_default_image(self) -> str:
        return "daytonaio/ai-test:0.1.0"  # todo: replace with ts image

    def get_run_command(self, code: str) -> str:
        base64_code = base64.b64encode(code.encode()).decode()
        return f"python3 -c \"exec(__import__('base64').b64decode('{base64_code}').decode())\""
